src/client/api/init.c

```diff
@@ -157,11 +157,21 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(out_debug, rc);
 
+	/** set up agent */
+	rc = dc_agent_init();
+	if (rc != 0)
+		D_GOTO(out_hhash, rc);
+
+	/** get CaRT configuration */
+	rc = dc_mgmt_net_cfg(NULL);
+	if (rc != 0)
+		D_GOTO(out_obj, rc);
```

Comment:
This is weird. I see mjmac asked about this already and you resolved.
But you moved out_obj at the end to after the pool, cont, placement, etc. why?
If the obj module is initialized here, why you assume the other modules are not initialized?

Reply:
sorry im looking at this more, but i still don't understand why this is not out_hhash, same for the daos_eq_lib_init()

Reply:
It's an error.  I looked at it so many times and apparently made the same mistake each time I looked at it.  dc_mgmt_net_cfg() should cleanup with out_hhash as should daos_eq_lib_init().